objcgen/helpers/objcgenerator-helpers.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+using IKVM.Reflection;
+using Type = IKVM.Reflection.Type;
+
+using Embeddinator;
+namespace ObjC {
+	public partial class ObjCGenerator {
+
+		public static string GetObjCConstructorName (ParameterInfo [] parameters)
+		{
+			StringBuilder name = new StringBuilder ("init");
+			foreach (var p in parameters) {
+				if (name.Length == 4)
+					name.Append ("With");
+				else
+					name.Append (' ');
+				name.Append (PascalCase (p.Name));
+				name.Append (":(").Append (GetTypeName (p.ParameterType)).Append (") ").Append (p.Name);
+			}
+			return name.ToString ();
+		}
+
+		public List<string> GetUnavailableParentCtors (Type type, List<ConstructorInfo> typeCtors)
+		{
+			var finalList = new List<string> ();
+			var baseType = type.BaseType;
+			if (baseType.Namespace == "System" && baseType.Name == "Object")
+				return finalList;
```

Comment:
Yes that was expected, I did not want to return`null` for collections IIRC is a bad practice. That being said I totally forgot about `Enumerable.Empty<T>` will change to that.